File: src/main/kotlin/org/ahant/validator/validation/DefaultFieldValidator.java
package org.ahant.validator.validation;

import com.google.common.collect.ImmutableSet;

import java.util.Set;

/**
 * Created by ahant on 8/14/2016.
 */
class DefaultFieldValidator implements FieldValidator<Object> {
    @Override
    public Set<String> validate(Object input) {
        return null != input ? null : ImmutableSet.of("null input");
    }
}

File: src/main/kotlin/org/ahant/validator/validation/DefaultFieldValidator.kt
package org.ahant.validator.validation

import com.google.common.collect.ImmutableSet

/**
 * Created by ahant on 8/14/2016.
 */
internal class DefaultFieldValidator : FieldValidator<Any> {
    override fun validate(input: Any): Set<String> {
        return if (null != input) ImmutableSet.of() else ImmutableSet.of("null input")
    }
}
